### PR TITLE
TP 1

### PR DESCRIPTION
**Pull request by ccgauche**
Nom: Laurent Gaucheron

### CODE REVIEW

File: 1_OCaml/TP/TP1.ipynb [hunk 1]
```diff
@@ -2,18 +2,10 @@
  "cells": [
```
Comment (fortierq) [inline 1_OCaml/TP/TP1.ipynb:2]: <p>Manque t-il le premier exercice ?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mp2i-fsm/mp2i-2021/pull/31/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (fortierq) [inline 1_OCaml/TP/TP1.ipynb:2]: <p>D'où vient cette formule ? Je pense que tu as fait une erreur.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mp2i-fsm/mp2i-2021/pull/31/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (fortierq) [inline 1_OCaml/TP/TP1.ipynb:2]: <p>Bien</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mp2i-fsm/mp2i-2021/pull/31/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (fortierq) [inline 1_OCaml/TP/TP1.ipynb:2]: <p>Preuve mathématique ?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mp2i-fsm/mp2i-2021/pull/31/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (fortierq) [inline 1_OCaml/TP/TP1.ipynb:2]: <p>Oui !</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mp2i-fsm/mp2i-2021/pull/31/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Comment (fortierq) [inline 1_OCaml/TP/TP1.ipynb:2]: <p>Très bien</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mp2i-fsm/mp2i-2021/pull/31/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

**Comment by fortierq** [comment]
Très bon travail

File: 1_OCaml/TP/TP1.ipynb (hunk 1 above)
Comment (ccgauche) [reply 1_OCaml/TP/TP1.ipynb:2]:
```ocaml
let tanh x = (let ex = exp x in 
              (ex -. (1. /. ex)) /. (ex +. (1. /. ex)));;
```
Pardon il je me suis trompé j'ai cru que une simplification était possible.

Comment (ccgauche) [reply 1_OCaml/TP/TP1.ipynb:2]: <p>Pardon je n'avais pas vu:</p><p><code>float  -&gt; float -&gt; float</code></p><p><code>(int -&gt; 'a) -&gt; int -&gt; 'a</code></p><p><code>(float -&gt; int) -&gt; ('a -&gt; float) -&gt; 'a -&gt; int</code></p><p> </p><p><code>f 1 2</code></p><p><code>g&nbsp;(fun&nbsp;x&nbsp;-&gt;&nbsp;2&nbsp;*&nbsp;x)&nbsp;1;;</code></p><p><code>h&nbsp;int_of_float&nbsp;(fun&nbsp;x&nbsp;-&gt;&nbsp;3.&nbsp;*.&nbsp;x)&nbsp;1.;;</code></p>

Comment (ccgauche) [reply 1_OCaml/TP/TP1.ipynb:2]: <p>1..</p><p>2^{4k} mod 5 = 16^k mod 5</p><p>or 16 mod 5 = 1</p><p>donc 16^k mod 5 = 1^k mod 5</p><p>soit 16^k mod 5 = 1.</p><p>2..</p><p>1357 mod 5 = 2</p><p>1357^2013 mod 5 = 2^(4*503+1) mod 5</p><p>1357^2013 mod 5 = 2*2^(4*503) mod 5</p><p>1357^2013 mod 5 = 2*1 mod 5</p><p>1357^2013 mod 5 = 2</p>

**Pull request closed without merging**